File: backend/ride-service/Sources/App/DTOs/GetRideOverviewDTO.swift
import Foundation

public struct GetRideOverviewDTO: Codable {
    public var id: UUID?
    public var name: String
    public var description: String?
    public var starts: Date
    public var latitude: Float
    public var longitude: Float
    
    public init(id: UUID? = nil, name: String, description: String? = nil, starts: Date, latitude: Float, longitude: Float) {
        self.id = id
        self.name = name
        self.description = description
        self.starts = starts
        self.latitude = latitude
        self.longitude = longitude
    }
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/RideServiceDTOs/GetRideOverviewDTO.kt
package net.ipv64.kivop.dtos.RideServiceDTOs

import java.time.LocalDateTime
import java.util.UUID

data class GetRideOverviewDTO(
    var id: UUID?,
    var name: String,
    var description: String?,
    var starts: LocalDateTime,
    var latitude: Float,
    var longitude: Float,
)
